packages/flutter_tools/lib/src/build_runner/web_fs.dart

```diff
@@ -272,12 +272,29 @@ class WebFs {
         return Response.ok(file.readAsBytesSync(), headers: <String, String>{
           'Content-Type': 'text/javascript',
         });
+      } else if (request.url.path.endsWith('dart_sdk.js.map')) {
+        final File file = fs.file(fs.path.join(
+          artifacts.getArtifactPath(Artifact.flutterWebSdk),
+          'kernel',
+          'amd',
+          'dart_sdk.js.map',
+        ));
+        return Response.ok(file.readAsBytesSync());
       } else if (request.url.path.endsWith('.dart')) {
+        print(request.url.path);
         // This is likely a sourcemap request. The first segment is the
         // package name, and the rest is the path to the file relative to
         // the package uri. For example, `foo/bar.dart` would represent a
         // file at a path like `foo/lib/bar.dart`. If there is no leading
         // segment, then we assume it is from the current package.
+
+        // Handle sdk requests have mangled urls from engine build.
```

Comment:
Something is wrong in the comment (or I'm misreading). Did you mean "requests that have"?

Reply:
Fixed